File: Hijeck/app/src/main/java/com/example/hijeck/DatabaseHelper.java
package com.example.hijeck;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.arch.core.executor.DefaultTaskExecutor;

public class DatabaseHelper extends SQLiteOpenHelper {
    private static String  name;
    public DatabaseHelper(Context context)
    {
        super(context,"Login.db",null,1);
    }
    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL("Create table userDetails(name text primary key,password text)");
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("drop table if exists userDetails");
    }
    //inserting database
    public boolean insert(String name ,String password)
    {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues contentValues = new ContentValues();
        contentValues.put("name",name);
        contentValues.put("password",password);
        long ins = db.insert("userDetails",null,contentValues);
        if(ins==-1)return false;
        else return true;
    }
    public Boolean checkName(String name)
    {
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor= db.rawQuery("select * from userDetails where name=?",new String[]{name});
        if(cursor.getCount()>0)return  false;
        else return true;

    }
    //checking
    public boolean emailpasscheck(String name,String pass)
    {
        DatabaseHelper.name=name;
        SQLiteDatabase db=this.getReadableDatabase();
        Cursor cursor=db.rawQuery("select * from userDetails where name=? and password=?",new String[]{name,pass});
        if(cursor.getCount()>0) return true;
        else return false;
    }

    public static String getName() {
        return name;
    }

}

File: GoogleSummerOfCode/app/src/main/java/com/example/googlesummerofcode/DatabaseHelper.kt
package com.example.googlesummerofcode

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper


class DatabaseHelper(context: Context?) :
    SQLiteOpenHelper(context, "Login.db", null, 1) {

      var name: String? = null

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("Create table userDetails(name text primary key,password text)")
    }

    override fun onUpgrade(
        db: SQLiteDatabase,
        oldVersion: Int,
        newVersion: Int
    ) {
        db.execSQL("drop table if exists userDetails")
    }

    //inserting database
    fun insert(name: String?, password: String?): Boolean {
        val db = this.writableDatabase
        val contentValues = ContentValues()
        contentValues.put("name", name)
        contentValues.put("password", password)
        val ins = db.insert("userDetails", null, contentValues)
        return ins != -1L
    }

    fun checkName(name: String): Boolean {
        val db = this.readableDatabase
        val cursor =
            db.rawQuery("select * from userDetails where name=?", arrayOf(name))
        return if (cursor.count > 0) false else true
    }

    //checking
    fun emailpasscheck(name: String?, pass: String?): Boolean {
         this.name = name
        val db = this.readableDatabase
        val cursor = db.rawQuery(
            "select * from userDetails where name=? and password=?",
            arrayOf(name, pass)
        )
        return cursor.count > 0
    }

}